### PR TITLE
Devicetree doc improvements 2022 04

### PR DESCRIPTION
**Pull request by mbolivar-nordic**

### CODE REVIEW

**Comment by mbolivar-nordic** [comment]
@MaureenHelm if you have any suggestions for the 3rd patch in particular, I'd appreciate them. I'm trying to avoid going around and around on how `default` is used, and sensor driver devicetree bindings reviews are the place where I have to do that the most. This docs PR is in attempt to make this easier on contributors.

File: doc/build/dts/bindings.rst [hunk 1]
```diff
@@ -1048,6 +1025,85 @@ The following general rules apply to vendor prefixes in :ref:`compatible
...
+safely defaulted. Candidates for default values include:
+
+- delays that would be different only under unusual conditions
+  (such as intervening hardware)
+- configuration for devices that have a standard initial configuration (such as
+  a USB audio headset)
+- defaults which match the vendor-specified power-on reset value
+  (as long as they are independent from other properties)
+
+Power-on reset values may be used for defaults as long as they're independent.
```
Comment (MaureenHelm) [inline doc/build/dts/bindings.rst:1059]: These two sentences are redundant.

I'm also not sure we should qualify with "as long as they're independent". I think power-on reset is sufficient and requiring independence may confuse people. Can you give an example of some properties that you'd advise against defining a default due to a dependency?

Comment (gmarull) [inline doc/build/dts/bindings.rst:1059]:
```suggestion
- defaults which match the vendor-specified power-on reset value
  (as long as they are independent from other properties)
```

(duplicated info)

**Comment by carlescufi** [comment]
@mbolivar-nordic can you rebase this please?

**Comment by mbolivar-nordic** [comment]
> @mbolivar-nordic can you rebase this please?

@carlescufi done

File: doc/build/dts/bindings.rst (hunk 1 above)
Comment (gmarull) [reply doc/build/dts/bindings.rst:1065]: ping ^

**Comment by mbolivar-nordic** [comment]
> Can you give an example of some properties that you'd advise against defining a default due to a dependency?

@MaureenHelm this rule was originally suggested by Peter Bigot. If we have

```
properties:
   foo: ...
   bar: ...
```

And changing `foo` from its power on reset value can only be done safely with an accompanying change in `bar`, then his argument was that making them both `required` forces the user to make a conscious choice, whereas allowing defaults for both `foo` and `bar` would allow the user to change `foo` without also having to change `bar` at build time, resulting in an incorrect configuration.

I find the argument convincing but don't know of any hardware like that off the top of my head.

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**